src/flow-timeout.c

```diff
@@ -88,6 +88,17 @@ static inline Packet *FlowForceReassemblyPseudoPacketSetup(Packet *p,
     p->flags |= PKT_STREAM_EOF;
     p->flags |= PKT_HAS_FLOW;
     p->flags |= PKT_PSEUDO_STREAM_END;
+    p->livedev = f->livedev;
+
+    if (f->vlan_id[0]) {
```

Comment:
This won’t work for vlan 0, which is why an explicit vlan index is used in other places. #3624 adds that index to the flow struct and I think it would be better to copy it back to the pseudo packet as well.

Reply:
You are right, fixing this in new PR. Thanks!